### PR TITLE
GWC Azure blob community module

### PR DESCRIPTION
**Pull request by aaime**
## Description

GWC Azure blobstore integration. Do not merge yet, requires https://github.com/GeoWebCache/geowebcache/pull/775 to be merged before. But if you want to review, feel free to do so.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)

The following are required only for core and extension modules, while they are welcomed, but not required, for community modules:
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: src/community/gwc-azure-blob/src/main/java/org/geoserver/gwc/web/blob/AzureBlobStorePanel.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+				
+			<label for="proxyUsername"><wicket:message key="proxyUsername" /></label>
+			<input id="proxyUsername" wicket:id="proxyUsername" type="text" class="text" />
+				
+			<label for="proxyPassword"><wicket:message key="proxyPassword" /></label>
+			<input id="proxyPassword" wicket:id="proxyPassword" type="text" class="text" />
+		</fieldset>
+	</wicket:panel>
+</body>
+</html>
```
Comment (bradh) [inline src/community/gwc-azure-blob/src/main/java/org/geoserver/gwc/web/blob/AzureBlobStorePanel.html]: Newline at end of file if you're going to make other changes.

File: src/community/gwc-azure-blob/src/main/resources/GeoServerApplication.properties [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#Generated by Eclipse Messages Editor (Eclipse Babel)
+
+AzureBlobStorePanel.container = Container *
+AzureBlobStorePanel.container.title = The name of the Azure container where the tiles are stored.
+AzureBlobStorePanel.accountName = Account name*
+AzureBlobStorePanel.accountName.title = The Azure Storage Account Name.
+AzureBlobStorePanel.accountKey = Account Key *
+AzureBlobStorePanel.accountKey.title = The Azure Storage Account Key.
+AzureBlobStorePanel.prefix = Azure Object Key Prefix
+AzureBlobStorePanel.prefix.title = A prefix path to use as the root to store tiles under the bucket.
```
Comment (bradh) [inline src/community/gwc-azure-blob/src/main/resources/GeoServerApplication.properties]: Update to match whatever you decided to call the storage / container.

File: src/community/release/ext-gwc-azure-blob.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		<include>*azure-*.jar</include>
+		<include>client-runtime*.jar</include>
+		<include>netty*.jar</include>
+		<include>rxjava*.jar</include>
+		<include>reactive-streams*.jar</include>
+		<include>joda-time*.jar</include>
+    <include>jackson*.jar</include>
+    <include>woodstox*.jar</include>
+    <include>*stax2*.jar</include>
+      </includes>
```
Comment (bradh) [inline src/community/release/ext-gwc-azure-blob.xml]: The formatting looks off here.

File: src/community/release/pom.xml [hunk 4]
```diff
@@ -130,6 +130,11 @@
       <groupId>org.geoserver.community</groupId>
       <artifactId>gs-gwc-s3</artifactId>
      <version>${project.version}</version>
+   </dependency>
+    <dependency>
```
Comment (bradh) [inline src/community/release/pom.xml]: Formatting off by 1?

File: doc/en/user/source/community/gwc-azure-blob/index.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+~~~~~~~~~~~~
+Azure storage account name
+
+Account key
+~~~~~~~~~~~
+Azure storage Account Key.
+
+Azure Object Key Prefix
+~~~~~~~~~~~~~~~~~~~~~~~
+A prefix path to use as the root to store tiles under the bucket (optional).
```
[the remark this answers is not in the document]
Comment (aaime) [reply doc/en/user/source/community/gwc-azure-blob/index.rst]: Yep

File: src/community/gwc-azure-blob/src/main/java/org/geoserver/gwc/web/blob/AzureBlobStorePanel.html (hunk 1 above)
Comment (aaime) [reply src/community/gwc-azure-blob/src/main/java/org/geoserver/gwc/web/blob/AzureBlobStorePanel.html]: Done

File: src/community/gwc-azure-blob/src/main/resources/GeoServerApplication.properties (hunk 2 above)
Comment (aaime) [reply src/community/gwc-azure-blob/src/main/resources/GeoServerApplication.properties]: Container

File: src/community/release/pom.xml (hunk 4 above)
Comment (aaime) [reply src/community/release/pom.xml]: Fixed

**Comment by aaime** [comment]
You're right, added now.

**Pull request merged**